## Q1. (approach 1)

In [1]:
import pandas as pd
data = pd.DataFrame({'Pat':[5, 4, 4, 3, 9, 4],
'Jack':[4, 8, 7, 5, 1, 5],
'Alex':[9, 9, 8, 10, 4, 10]})
data

Pat  Jack  Alex
0    5     4     9
1    4     8     9
2    4     7     8
3    3     5    10
4    9     1     4
5    4     5    10

In [2]:
data_new = data.melt(id_vars=None , value_vars=['Pat','Jack','Alex'])
data_new

variable  value
0       Pat      5
1       Pat      4
2       Pat      4
3       Pat      3
4       Pat      9
5       Pat      4
6      Jack      4
7      Jack      8
8      Jack      7
9      Jack      5
10     Jack      1
11     Jack      5
12     Alex      9
13     Alex      9
14     Alex      8
15     Alex     10
16     Alex      4
17     Alex     10

In [3]:
data_new.groupby('variable').mean()

value
variable          
Alex      8.333333
Jack      5.000000
Pat       4.833333

In [4]:
import statsmodels.api as sm
from  statsmodels.formula.api import ols

In [7]:
ln = ols('value ~ variable', data=data_new).fit()
table=sm.stats.anova_lm(ln)
table

df     sum_sq    mean_sq         F    PR(>F)
variable   2.0  46.777778  23.388889  4.488273  0.029668
Residual  15.0  78.166667   5.211111       NaN       NaN

### Approach 2

In [11]:
import pandas as pd

Pat = [5, 4, 4, 3, 9, 4]
Jack = [4, 8, 7, 5, 1, 5]
Alex = [9, 9, 8, 10, 4, 10]
all_scores = Pat + Jack + Alex
name = (['Pat']*len(Pat))+(['Jack']*len(Jack))+(['Alex']*len(Alex))

data1 = pd.DataFrame({'name_col':name , 'score':all_scores})
data1

name_col  score
0       Pat      5
1       Pat      4
2       Pat      4
3       Pat      3
4       Pat      9
5       Pat      4
6      Jack      4
7      Jack      8
8      Jack      7
9      Jack      5
10     Jack      1
11     Jack      5
12     Alex      9
13     Alex      9
14     Alex      8
15     Alex     10
16     Alex      4
17     Alex     10

In [17]:
data1.groupby('name_col').mean()

score
name_col          
Alex      8.333333
Jack      5.000000
Pat       4.833333

In [18]:
import statsmodels.api as sm
from statsmodels.formula.api import ols

In [19]:
lns=ols('score ~ name_col',data=data1).fit()
table1=sm.stats.anova_lm(lns)
table1

df     sum_sq    mean_sq         F    PR(>F)
name_col   2.0  46.777778  23.388889  4.488273  0.029668
Residual  15.0  78.166667   5.211111       NaN       NaN

From both the approaches we get the same answer . Now looking at the P value......since the P value 0.029668 is less than 0.05 hence we will reject the null hypothesis . This means that the MEANS of all the population are not equal.....hence one or more groups have different mean